--- a/utils/dialog/CMakeLists.txt
+++ b/utils/dialog/CMakeLists.txt
@@ -23,7 +23,7 @@ qt_add_qml_module(hyprland-dialog
 )

 target_link_libraries(hyprland-dialog PRIVATE
-    Qt6::Widgets Qt6::QuickControls2 Qt6::WaylandClientPrivate PkgConfig::hyprutils
+    Qt6::Widgets Qt6::QuickControls2 Qt6::WaylandClient PkgConfig::hyprutils
 )


--- a/utils/update-screen/CMakeLists.txt
+++ b/utils/update-screen/CMakeLists.txt
@@ -31,7 +31,7 @@ qt_add_resources(hyprland-update-screen "assets"
 )

 target_link_libraries(hyprland-update-screen PRIVATE
-    Qt6::Widgets Qt6::QuickControls2 Qt6::WaylandClientPrivate PkgConfig::hyprutils
+    Qt6::Widgets Qt6::QuickControls2 Qt6::WaylandClient PkgConfig::hyprutils
 )


--- a/utils/donate-screen/CMakeLists.txt
+++ b/utils/donate-screen/CMakeLists.txt
@@ -29,7 +29,7 @@ qt_add_resources(hyprland-donate-screen "assets"
 )

 target_link_libraries(hyprland-donate-screen PRIVATE
-    Qt6::Widgets Qt6::QuickControls2 Qt6::WaylandClientPrivate PkgConfig::hyprutils
+    Qt6::Widgets Qt6::QuickControls2 Qt6::WaylandClient PkgConfig::hyprutils
 )
